sys/kern/callout.c

```diff
@@ -62,9 +71,19 @@ static void callout_thread(void *arg) {
 
     /* Execute callout's function. */
     elem->c_func(elem->c_arg);
-    callout_clear_active(elem);
-    /* Wake threads that wait for execution of this callout in callout_drain. */
-    sleepq_broadcast(elem);
+
+    WITH_SPIN_LOCK (&ci.lock) {
+      callout_clear_active(elem);
+      if (elem->c_interval) {
+        /* Periodic callout: reschedule. */
+        elem->c_time += elem->c_interval;
+        _callout_schedule(elem);
```

Comment:
It shouldn't happen, but I have been thinking about a situation, when `elem->c_time + n * elem->c_interval < actual time` and we will schedule it n times knowing that the time already passed. Should we worry about such case?

Reply:
I don't think rescheduling to the nearest interval multiple in the future is inherently better than rescheduling to a missed deadline: to me it's a matter of choice.